src/Bicep.LangServer/Handlers/BicepEditLinterRuleCommandHandler.cs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using System;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+using Bicep.Core;
+using Bicep.Core.Configuration;
+using Bicep.LanguageServer.Configuration;
+using Bicep.LanguageServer.Telemetry;
+using MediatR;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using OmniSharp.Extensions.JsonRpc;
+using OmniSharp.Extensions.LanguageServer.Protocol;
+using OmniSharp.Extensions.LanguageServer.Protocol.Server;
+using OmniSharp.Extensions.LanguageServer.Protocol.Window;
+using OmniSharp.Extensions.LanguageServer.Protocol.Workspace;
+using Range = OmniSharp.Extensions.LanguageServer.Protocol.Models.Range;
+
+namespace Bicep.LanguageServer.Handlers
+{
+    /// <summary>
+    /// Handles the internal command for code actions to edit a particular linter rule in the bicepconfig.json file
+    /// </summary>
+    public class BicepEditLinterRuleCommandHandler : ExecuteTypedCommandHandlerBase<DocumentUri, string, string>
+    {
+        private readonly string DefaultBicepConfig;
+        private readonly ILanguageServerFacade server;
+        private readonly ITelemetryProvider telemetryProvider;
+
+        public BicepEditLinterRuleCommandHandler(ISerializer serializer, ILanguageServerFacade server, ITelemetryProvider telemetryProvider)
+            : base(LanguageConstants.EditLinterRuleCommandName, serializer)
+        {
+            DefaultBicepConfig = DefaultBicepConfigHelper.GetDefaultBicepConfig();
+            this.server = server;
+            this.telemetryProvider = telemetryProvider;
+        }
+
+        public override async Task<Unit> Handle(DocumentUri documentUri, string ruleCode, string bicepConfigFilePath, CancellationToken cancellationToken)
+        {
+            // bicepConfigFilePath will be empty string if no current configuration file was found
+            if (string.IsNullOrEmpty(bicepConfigFilePath))
+            {
+                // There is no configuration file currently - create one in the default location
+                var targetFolder = BicepGetRecommendedConfigLocationHandler.GetRecommendedConfigFileLocation(this.server, documentUri.GetFileSystemPath());
```

Comment:
Where is this used?

Reply:
whoops thanks

Reply:
Fixed the test that should have caught thisd7f40f6079e4d24a537b918a400d831278045b88